snap/local/start-mysql-router.sh

```diff
@@ -2,4 +2,4 @@
 
 # For security measures, daemons should not be run as sudo. Execute mysqlrouter as the non-sudo user: snap-daemon.
 exec $SNAP/usr/bin/setpriv --clear-groups --reuid snap_daemon \
-  --regid snap_daemon -- $SNAP/usr/bin/mysqlrouter "$@"
+  --regid root -- $SNAP/usr/bin/mysqlrouter --config $SNAP_DATA/etc/mysqlrouter/mysqlrouter.conf
```

Comment:
I tested a bootstrap of mysqlrouter and then started the mysqlrouter daemon without any issues. I believe that we use this pattern in all snap applications (daemon or not), so that the root user has access to the files in $SNAP_DATA or $SNAP_COMMON without providing all users access to these files (re: `user-group-others` permissions)